docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,213 @@
+  - [For Newcomers](#for-newcomers)
+- [How to Contribute](#how-to-contribute)
+  - [Set up your Local Development Environment](#set-up-your-local-development-environment)
+  - [Testing Changes](#testing-changes)
+  - [Signing-off on Commits](#signing-off-on-commits)
+    (#testing-your-changes)
+  - [Signing-off on Commits](#signing-off-on-commits)
+
+# Before You Get Started
+
+## Code of Conduct
+
+## For Newcomers
+
+# How to Contribute
+
+<!--
+## Prerequisites
+
+Make sure you have the following prerequisites installed on your operating system before you start contributing:
+
+- [Nodejs and npm](https://nodejs.org/en/)
+
+  To verify run:
+
+  ```
+  node -v
+  ```
+
+  ```
+  npm -v
+  ```
+
+- [Gatsby.js](https://www.gatsbyjs.com/)
+
+  To verify run:
+
+  ```
+  gatsby --version
+  ```
+
+**Note:** If you're on a _Windows environment_ then it is highly recommended that you install [Windows Subsystem for Linux (WSL)](https://docs.microsoft.com/en-us/windows/wsl/install) both for performance and ease of use. Refer to the [documentation](https://docs.microsoft.com/en-us/windows/dev-environment/javascript/gatsby-on-wsl) for the installation of _Gatsby.js on WSL_. -->
+
+## Set up your Local Development Environment
+
+Follow the following instructions to start contributing.
+
+**1.** Fork [this](https://github.com/devfile/devworkspace-operator) repository.
+
+**2.** Clone your forked copy of the project.
+
+```
+git clone https://github.com/<your-github-username>/devworkspace-operator.git
+```
+
+**3.** Navigate to the project directory.
+
+```
+cd devworkspace-operator
+```
+
+**4.** Add a reference(remote) to the original repository.
+
+```
+git remote add upstream https://github.com/devfile/devworkspace-operator.git
+```
+
+**5.** Check the remotes for this repository.
+
+```
+git remote -v
+```
+
+**6.** Always take a pull from the upstream repository to your master branch to keep it at par with the main project (updated repository).
+
+```
+git pull upstream master
+```
+
+**7.** Create a new branch.
+
+```
+git checkout -b <your_branch_name>
+```
+
+**8.** Start the minikube cluster.
+
+```
+minikube start
+```
+
+**9.** Enable the ingress add-on for your minikube cluster.
+
+```
+minikube addons enable ingress
+```
+
+**10.** Set the namespace environment variable for the development environment to avoid changes inside the default namespace.
+
+```
+export NAMESPACE="devworkspace-controller"
+```
+
+**11.** Install the kubernetes certificate management controller to generate and manage TLS certificates for your cluster.
+
+```
+make install cert-manager.
+```
+
+**12.** Install the dependencies for running the devworkspace-operator in your cluster.
+
+```
+make install
+```
+
+**13.** Scale down the replicas of pods to 0.
+
+```
+kubectl patch deployment/devworkspace-controller-manager --patch "{\"spec\":{\"replicas\":0}}" -n $NAMESPACE
+```
+
+**14.** Run the devworkspace-operator.
+
+```
+make run
+```
```

Comment:
Structurally, this should appear in a section like "running the controller locally"; this is not the only way to contribute so we shouldn't present it as such.